src/ALCProxy.Communication/CommObjects.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. 
+// The .NET Foundation licenses this file to you under the MIT license. 
+// See the LICENSE file in the project root for more information. 
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization;
+
+namespace ALCProxy.Communication
+{
+    public class ClientDispatch : ALCClient
+    {
+        public ClientDispatch(Type intType) : base(intType, "ServerDispatch`1") { }
+        protected override object SerializeParameter(object param, Type paramType)
+        {
+            MemoryStream stream = new MemoryStream();
+            //Serialize the Record object to a memory stream using DataContractSerializer.  
+            DataContractSerializer serializer = new DataContractSerializer(paramType);
+            serializer.WriteObject(stream, param);
+            return stream;
+
+        }
+        protected override object DecryptReturnType(object s, Type returnType)
```

Comment:
I'd suggest `Deserialize` instead of `Decrypt`

Reply:
Fixed.